src/tools/generate/generate_chapters.js

```diff
@@ -51,6 +51,9 @@ const parse_file = async (markdown,chapter) => {
   const html = converter.makeHtml(markdown);
   let body = html;
 
+  // Make headers links
+  const header_regex = /<h([2-6]) id="(.*?)">(.*?)<\/h[2-6]>/gi;
```

Comment:
Will this work if there are other attributes in the header?

Reply:
The headers are generated from markdown so currently don't have any other attributes other than the id (which is auto generated by showndown).

In theory they could if someone uses HTML for the headers, but would rather keep the regex simple that complicate it with before and after bits for the id, so I say we leave as is and revisit it we need to.

WDYT?

Reply:
I added a bit more to this to make sure we at least don't delete attributes:

```javascript
  const header_regex = /<h([2-6]) id="(.*?)"(.*?)>(.*?)<\/h[2-6]>/gi;
  return html.replace(header_regex, '<h$1 id="$2"$3><a href="#$2" class="anchor-link">$4</a></h$1>');
```